### PR TITLE
Fixes the leave a review option in about screen 

### LINKED ISSUE #499: Change the behaviour of 'Leave a review' under About screen
#### Software and hardware versions 
Collect v1.4, Android v4.4, device used...

#### Problem description
In `About screen` there is a `Leave a review` option. The description of the option says _Leave a review on Google Play_. On clicking the option a chooser intent appears. The problem is that if the user has google play store installed, then also a list of all available options appear instead of directly opening the Google Play Store app.

#### Steps to reproduce the problem
Click **Leave a review** under `About screen`
#### Expected behavior
If the device has GooglePlayStore installed then directly open GooglePlayStore. Otherwise ask the user to choose from a list of available options

### PR DESCRIPTION
**Pull request by shobhitagarwal1612**
Fixes #499 

- Also replaced the deprecated methods(addPreferenceFromResource)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/preferences/AboutPreferencesActivity.java [hunk 1]
```diff
@@ -16,55 +16,114 @@
...
+
+                case KEY_LEAVE_A_REVIEW:
+                    try {
+                        // Open the google play store app if present
+                        intent = new Intent(Intent.ACTION_VIEW,
+                                Uri.parse("market://details?id=" + APP_PACKAGE_NAME));
+                        PackageManager packageManager = getActivity().getPackageManager();
+                        List<ResolveInfo> list = packageManager.queryIntentActivities(intent, 0);
+
+                        for (int a = 0; a < list.size(); a++) {
```
Comment (dcbriccetti) [inline collect_app/src/main/java/org/odk/collect/android/preferences/AboutPreferencesActivity.java]: Would foreach work here? `for (ResolveInfo info : list)`

Comment (shobhitagarwal1612) [reply collect_app/src/main/java/org/odk/collect/android/preferences/AboutPreferencesActivity.java]: @dcbriccetti Yeah, it looks neater than before. Thanks :smiley:

**Comment by lognaturel** [comment]
When I click the button in my Nexus 5 API 16 or API 22 emulator without Google Play Store, nothing happens! There is a browser and that's what it was automatically opened in before. It works great on my physical device with the play store.

**Comment by shobhitagarwal1612** [comment]
@lognaturel Thanks for pointing out the issue. The problem was that the exception was not getting raised as the intent was only occurring in case google play store was found. I have fixed the bug and have also tested the app on devices not having google play store.

**Comment by lognaturel** [comment]
Great, thanks for the clear explanation of the fix!

Next time you add to or make reviews to a feature, just go ahead and submit the changes as a new commit. That makes review easier, documents the changes and doesn't affect master's history because I will squash the commits when merging.

**Comment by shobhitagarwal1612** [comment]
I will keep in mind not to squash the commits from now on

**Pull request merged**